tools/autograd/gen_variable_type.py

```diff
@@ -717,6 +723,7 @@ def emit_profiled_body(declaration):
     modifies_arguments = inplace or is_out_fn
     returns_void = len(returns) == 0
 
+    base_name = name[:-1] if inplace else name[:-4] if is_out_fn else name
```

Comment:
You should already have access to the base name of the function without doing this in this file. Something like `declaration["base_name"]` ?